test/parallel/test-buffer-tojson.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const Buffer = require('buffer').Buffer;
+
+{
+  assert.deepStrictEqual(JSON.stringify(Buffer.alloc(0)),
+                         '{"type":"Buffer","data":[]}');
+  assert.deepStrictEqual(JSON.stringify(Buffer.from([1, 2, 3, 4])),
+                         '{"type":"Buffer","data":[1,2,3,4]}');
+}
+
+// issue GH-7849
+{
+  const buf = Buffer.from('test');
+  const json = JSON.stringify(buf);
+  const obj = JSON.parse(json);
+  const copy = Buffer.from(obj);
+
+  assert(buf.equals(copy));
```

Comment:
I think `assert.deepStrictEqual(buf, copy)` does the job too(uses `compare` underneath).